src/routes/safe/components/Settings/SpendingLimit/Amount.tsx

```diff
@@ -0,0 +1,55 @@
+import { TextField } from '@gnosis.pm/safe-react-components'
+import React from 'react'
+import { useField } from 'react-final-form'
+import { useSelector } from 'react-redux'
+
+import GnoField from 'src/components/forms/Field'
+import { composeValidators, minValue, mustBeFloat, required } from 'src/components/forms/validator'
+import { extendedSafeTokensSelector } from 'src/routes/safe/container/selector'
+import styled from 'styled-components'
+import { useStyles } from './style'
+
+export const Field = styled(GnoField)`
+  margin: 8px 0;
+  width: 100%;
+`
+
+const AmountInput = styled.div`
+  grid-area: amountInput;
+`
+
+const GnoTextField = styled(TextField)`
+  margin: 0;
+`
+
+export const Amount = (): React.ReactElement => {
+  const classes = useStyles()
+
+  const {
+    input: { value: tokenAddress },
+  } = useField('token', { subscription: { value: true } })
+  const {
+    meta: { touched, visited },
+  } = useField('amount', { subscription: { touched: true, visited: true } })
+
+  const tokens = useSelector(extendedSafeTokensSelector)
+
+  const selectedTokenRecord = tokens.find((token) => token.address === tokenAddress)
+
+  const validate = (touched || visited) && composeValidators(required, mustBeFloat, minValue(0, false))
+
+  return (
+    <AmountInput>
+      <Field
```

Comment:
same here, Input field form safe-react-component has support for react-final-form, perhaps you could use it instead of GnoField.

Reply:
Actually, if you check it's using the SRC's `TextField` but there's a mess with the names that I'll fix right away.

Reply:
On the other hand, `GnoField` is a wrapper around `react-final-form`'s `Field` component. I'm not sure why would we need RFF support in the SRC library.